src/langauge-servers/configStore.ts

```diff
@@ -187,43 +187,38 @@ class LangServerConfigStore {
         }
     }
 
+    private isPythonLSInstalled() {
+        const pipLocation = getPipLocationFromPythonDirectory(this.serverLocations["python"]);
+        const { stdout, stderr } = cp.spawnSync(
+            PIP_EXEC_FILENAME,
+            ["show", "python-lsp-server"],
+            { cwd: pipLocation, encoding: "utf-8" }
+        );
+        const trimmedError = _.trim(stderr);
+        const trimmedOutput = _.trim(stdout);
+        return _.includes(trimmedOutput, "Name: python-lsp-server") && !_.startsWith(trimmedError, "WARNING: Package(s) not found:");
```

Comment:
Is there no better way to check it than hardcoded strings? I see it breaking in the future

Reply:
> Is there no better way to check it than hardcoded strings? I see it breaking in the future

@ron-rookout The status code is 1 I believe, but this could change too.
I just looked through pip's source code, it does not appear like there is any other indicator.
Would you rather I just check if stdout is not empty (thus the command has a success status)?

Reply:
> If a status code 1 is enough to determine that's the error you're looking for, wouldn't it be enough to check it? Or are there other errors with status code 1 that are irrelevant?

I believe there could be other errors. However, if there is an error, perhaps we should consider it as if we could not find an installation, thus returning false?

Reply:
The question is what is the message displayed to the user. If the user has it installed and there's another error, would they still get something along the lines as "The language server isn't installed"?

Reply:
> The question is what is the message displayed to the user. If the user has it installed and there's another error, would they still get something along the lines as "The language server isn't installed"?

I think the stderr from pip install operation (if this one failed too) will be displayed.

Reply:
If we can just display the stderr then no problem, just check for an error in general, not a specific one. I think.

Reply:
> If we can just display the stderr then no problem, just check for an error in general, not a specific one. I think.

Ok then :)

Reply:
@ron-rookout Done